ruby/vampires.rb

```diff
@@ -1,15 +1,58 @@
-puts "What is your name?"
-name = gets.chomp
-puts "How old are you?"
-age = gets.chomp
-puts "What year were you born?"
-year = gets.chomp
-puts "Our company serves garlic bread. Should we order some for you (y/n)"
-order = gets.chomp
-puts "Would you like to enroll in the company's health insurance? (y/n)"
-insurance = gets.chomp
-
-puts case
-	when name == ("Drake Cula" || "Tu Fang")
-		puts "Definitely a vampire."
-end 
+#Repeating Loop
+puts "How many employees will be processed?"
+repeat = gets.chomp.to_i
+repeat.times do
+
+	#Initial Questionnaire
+	puts
+	puts "What is your name?"
+	name = gets.chomp
+	puts "How old are you?"
+	age = gets.to_i
+	puts "What year were you born?"
+	year = gets.to_i
+	puts "Our company serves garlic bread. Should we order some for you? (y/n)"
+	order = gets.chomp.downcase
+	ordered_garlic = order == "y"
+	puts "Would you like to enroll in the company's health insurance? (y/n)"
+	insurance = gets.chomp.downcase
+	enroll_insurance = insurance == "y"
+
+	correct_age = (age == 2017 - year)
+
+	#Analyzing Results
+	result = "Results inconclusive"
+	if correct_age && (ordered_garlic || enroll_insurance)
+		result = "Probably not a vampire."
+	end
+
+	if (correct_age == false) && (!ordered_garlic || !enroll_insurance)
+		result = "Probably a vampire."
+	end
+
+	if (correct_age == false) && (!ordered_garlic) && (!enroll_insurance)
+		result = "Almost certainly a vampire."
+	end 
+
+	if name == "Drake Cula" || name == "Tu Fang"
+		result = "Definitely a vampire."
+	end
+
+	#Allergy Survey
+	allergy = ""
+	puts "List all allergies that you have. Enter 'done' when you are finished."
+	until allergy == "done"

```

Comment:
This section should happen before the final vampire evaluations.  Try to keep your logic organized in a way that the program collects all of the necessary data first and then does a full evaluation.

Collecting the allergy data first and then running returning the vampire results all at once is a better practice vs collecting some of the data, returning an eval and then re-evaluating.